From 3a97da9401ae245e8a604af5e1e0dbb044569f0a Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Fri, 25 Sep 2026 17:08:13 -0400
Subject: [PATCH] UHDSoapyDevice: define get_stream_info for UHD>=4.11 (#78)

Upstream commit pothosware/SoapyUHD@3a97da9, backported onto the revision
pinned by nixpkgs (cf78b9ca3bddfc9263d2acb7e8afcb0036938163).

UHD 4.11 made uhd::rx_streamer::get_stream_info() and
uhd::tx_streamer::get_stream_info() pure virtual, so UHDSoapyRxStream and
UHDSoapyTxStream (which derive from them) became abstract and could not be
instantiated.

Signed-off-by: Ryan Volz <ryan.volz@gmail.com>
---
 UHDSoapyDevice.cpp | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/UHDSoapyDevice.cpp b/UHDSoapyDevice.cpp
--- a/UHDSoapyDevice.cpp
+++ b/UHDSoapyDevice.cpp
@@ -716,6 +716,13 @@ public:
     }
     #endif
 
+    #if UHD_VERSION >= 4110000
+    uhd::device_addr_t get_stream_info(const size_t /*chan*/) const override
+    {
+        throw uhd::not_implemented_error("get_stream_info is not implemented here!");
+    }
+    #endif
+
 private:
     SoapySDR::Device *_device;
     SoapySDR::Stream *_stream;
@@ -868,6 +875,13 @@ public:
     }
     #endif
 
+    #if UHD_VERSION >= 4110000
+    uhd::device_addr_t get_stream_info(const size_t /*chan*/) const override
+    {
+        throw uhd::not_implemented_error("get_stream_info is not implemented here!");
+    }
+    #endif
+
 private:
     bool _active;
     SoapySDR::Device *_device;
